[Bug] Internal Service Error OIDC via Authentik

**RomM version**
3.7.2

**Describe the bug**
Logging in via authentik will cause a Internal Service Error, attempting to login again seems to work.

**To Reproduce**
Steps to reproduce the behavior:

1. Go to login.
2. Click on authentik login
3. Login via authentik
4. See error

**Expected behavior**
Login with authentik without error.

**Additional context**



[2025-01-12 16:44:30 +0000] [23] [ERROR] Exception in ASGI application
Traceback (most recent call last):
  File "/src/.venv/lib/python3.12/site-packages/uvicorn/protocols/http/h11_impl.py", line 407, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/uvicorn/middleware/proxy_headers.py", line 69, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 409, in _sentry_patched_asgi_app
    return await middleware(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/asgi.py", line 161, in _run_asgi3
    return await self._run_app(scope, receive, send, asgi_version=3)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/asgi.py", line 262, in _run_app
    raise exc from None
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/asgi.py", line 257, in _run_app
    return await self.app(
           ^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/applications.py", line 113, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/errors.py", line 187, in __call__
    raise exc
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/errors.py", line 165, in __call__
    await self.app(scope, receive, _send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 185, in __call__
    with collapse_excgroups():
         ^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/contextlib.py", line 158, in __exit__
    self.gen.throw(value)
  File "/src/.venv/lib/python3.12/site-packages/starlette/_utils.py", line 82, in collapse_excgroups
    raise exc
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 187, in __call__
    response = await self.dispatch_func(request, call_next)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/backend/utils/context.py", line 41, in set_context_middleware
    return await call_next(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 163, in call_next
    raise app_exc
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 149, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/backend/handler/auth/middleware.py", line 147, in __call__
    await self.app(scope, receive, send_wrapper)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 348, in _sentry_authenticationmiddleware_call
    await old_call(self, scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/authentication.py", line 48, in __call__
    await self.app(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/backend/handler/auth/middleware.py", line 19, in __call__
    await super().__call__(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/starlette_csrf/middleware.py", line 72, in __call__
    await self.app(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/cors.py", line 85, in __call__
    await self.app(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 298, in _sentry_exceptionmiddleware_call
    await old_call(self, scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 715, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 735, in app
    await route.handle(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 73, in app
    response = await f(request)
               ^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/fastapi.py", line 143, in _sentry_app
    return await old_app(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/fastapi/routing.py", line 301, in app
    raw_response = await run_endpoint_function(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/fastapi/routing.py", line 212, in run_endpoint_function
    return await dependant.call(**values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/backend/endpoints/auth.py", line 254, in auth_openid
    token = await oauth.openid.authorize_access_token(request)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/authlib/integrations/starlette_client/apps.py", line 80, in authorize_access_token
    params = self._format_state_params(state_data, params)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/authlib/integrations/base_client/sync_app.py", line 234, in _format_state_params
    raise MismatchingStateError()
authlib.integrations.base_client.errors.MismatchingStateError: mismatching_state: CSRF Warning! State not equal in request and response.

Posting logs, I have the same stacktrace but actually a different error:
`ValueError: Invalid key set format`


```
[2025-01-21 03:51:43 +1300] [24] [ERROR] Exception in ASGI application
Traceback (most recent call last):
  File "/src/.venv/lib/python3.12/site-packages/uvicorn/protocols/http/h11_impl.py", line 407, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/uvicorn/middleware/proxy_headers.py", line 69, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 409, in _sentry_patched_asgi_app
    return await middleware(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/asgi.py", line 161, in _run_asgi3
    return await self._run_app(scope, receive, send, asgi_version=3)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/asgi.py", line 262, in _run_app
    raise exc from None
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/asgi.py", line 257, in _run_app
    return await self.app(
           ^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/applications.py", line 113, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/errors.py", line 187, in __call__
    raise exc
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/errors.py", line 165, in __call__
    await self.app(scope, receive, _send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 185, in __call__
    with collapse_excgroups():
         ^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/contextlib.py", line 158, in __exit__
    self.gen.throw(value)
  File "/src/.venv/lib/python3.12/site-packages/starlette/_utils.py", line 82, in collapse_excgroups
    raise exc
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 187, in __call__
    response = await self.dispatch_func(request, call_next)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/backend/utils/context.py", line 41, in set_context_middleware
    return await call_next(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 163, in call_next
    raise app_exc
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 149, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/backend/handler/auth/middleware.py", line 147, in __call__
    await self.app(scope, receive, send_wrapper)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 348, in _sentry_authenticationmiddleware_call
    await old_call(self, scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/authentication.py", line 48, in __call__
    await self.app(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/backend/handler/auth/middleware.py", line 19, in __call__
    await super().__call__(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/starlette_csrf/middleware.py", line 72, in __call__
    await self.app(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/cors.py", line 85, in __call__
    await self.app(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 298, in _sentry_exceptionmiddleware_call
    await old_call(self, scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 715, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 735, in app
    await route.handle(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 73, in app
    response = await f(request)
               ^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/fastapi.py", line 143, in _sentry_app
    return await old_app(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/fastapi/routing.py", line 301, in app
    raw_response = await run_endpoint_function(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/fastapi/routing.py", line 212, in run_endpoint_function
    return await dependant.call(**values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/backend/endpoints/auth.py", line 254, in auth_openid
    token = await oauth.openid.authorize_access_token(request)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/authlib/integrations/starlette_client/apps.py", line 84, in authorize_access_token
    userinfo = await self.parse_id_token(token, nonce=state_data['nonce'], claims_options=claims_options)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/authlib/integrations/base_client/async_openid.py", line 69, in parse_id_token
    key=JsonWebKey.import_key_set(jwk_set),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/authlib/jose/rfc7517/jwk.py", line 55, in import_key_set
    raise ValueError('Invalid key set format')
ValueError: Invalid key set format

```

I'm seeing presumably a similar error. Compose and error below. I'm not sure what request URL is requiring an http:// or https:// that is missing (last line of the logs).

```
2025-01-27T20:54:41.337940311Z [2025-01-27 12:54:41 -0800] [24] [ERROR] Exception in ASGI application
2025-01-27T20:54:41.337975942Z Traceback (most recent call last):
2025-01-27T20:54:41.338002044Z   File "/src/.venv/lib/python3.12/site-packages/uvicorn/protocols/http/h11_impl.py", line 407, in run_asgi
2025-01-27T20:54:41.338007315Z     result = await app(  # type: ignore[func-returns-value]
2025-01-27T20:54:41.338011859Z              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.338015920Z   File "/src/.venv/lib/python3.12/site-packages/uvicorn/middleware/proxy_headers.py", line 69, in __call__
2025-01-27T20:54:41.338024102Z     return await self.app(scope, receive, send)
2025-01-27T20:54:41.338028861Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.338033738Z   File "/src/.venv/lib/python3.12/site-packages/fastapi/applications.py", line 1054, in __call__
2025-01-27T20:54:41.338042840Z     await super().__call__(scope, receive, send)
2025-01-27T20:54:41.338047836Z   File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 409, in _sentry_patched_asgi_app
2025-01-27T20:54:41.338053243Z     return await middleware(scope, receive, send)
2025-01-27T20:54:41.338058823Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.338067382Z   File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/asgi.py", line 161, in _run_asgi3
2025-01-27T20:54:41.338076201Z     return await self._run_app(scope, receive, send, asgi_version=3)
2025-01-27T20:54:41.338082560Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.338105140Z   File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/asgi.py", line 262, in _run_app
2025-01-27T20:54:41.338111249Z     raise exc from None
2025-01-27T20:54:41.338116654Z   File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/asgi.py", line 257, in _run_app
2025-01-27T20:54:41.338122384Z     return await self.app(
2025-01-27T20:54:41.338145013Z            ^^^^^^^^^^^^^^^
2025-01-27T20:54:41.338154437Z   File "/src/.venv/lib/python3.12/site-packages/starlette/applications.py", line 113, in __call__
2025-01-27T20:54:41.338160677Z     await self.middleware_stack(scope, receive, send)
2025-01-27T20:54:41.338166085Z   File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
2025-01-27T20:54:41.338192140Z     return await old_call(app, scope, new_receive, new_send, **kwargs)
2025-01-27T20:54:41.338197266Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.338202268Z   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/errors.py", line 187, in __call__
2025-01-27T20:54:41.338219496Z     raise exc
2025-01-27T20:54:41.338226693Z   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/errors.py", line 165, in __call__
2025-01-27T20:54:41.338232861Z     await self.app(scope, receive, _send)
2025-01-27T20:54:41.338238605Z   File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
2025-01-27T20:54:41.338248506Z     return await old_call(app, scope, new_receive, new_send, **kwargs)
2025-01-27T20:54:41.338253578Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.338258717Z   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 185, in __call__
2025-01-27T20:54:41.338272988Z     with collapse_excgroups():
2025-01-27T20:54:41.338279235Z          ^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.338284220Z   File "/usr/lib/python3.12/contextlib.py", line 158, in __exit__
2025-01-27T20:54:41.338290132Z     self.gen.throw(value)
2025-01-27T20:54:41.338326840Z   File "/src/.venv/lib/python3.12/site-packages/starlette/_utils.py", line 82, in collapse_excgroups
2025-01-27T20:54:41.338335669Z     raise exc
2025-01-27T20:54:41.338340529Z   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 187, in __call__
2025-01-27T20:54:41.338345902Z     response = await self.dispatch_func(request, call_next)
2025-01-27T20:54:41.338354560Z                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.338357766Z   File "/backend/utils/context.py", line 41, in set_context_middleware
2025-01-27T20:54:41.338363396Z     return await call_next(request)
2025-01-27T20:54:41.338368459Z            ^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.338381796Z   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 163, in call_next
2025-01-27T20:54:41.338387838Z     raise app_exc
2025-01-27T20:54:41.338395105Z   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 149, in coro
2025-01-27T20:54:41.338400461Z     await self.app(scope, receive_or_disconnect, send_no_error)
2025-01-27T20:54:41.338413272Z   File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
2025-01-27T20:54:41.338419088Z     return await old_call(app, scope, new_receive, new_send, **kwargs)
2025-01-27T20:54:41.338424836Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.338437914Z   File "/backend/handler/auth/middleware.py", line 156, in __call__
2025-01-27T20:54:41.338443083Z     await self.app(scope, receive, send_wrapper)
2025-01-27T20:54:41.338447642Z   File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 348, in _sentry_authenticationmiddleware_call
2025-01-27T20:54:41.338463149Z     await old_call(self, scope, receive, send)
2025-01-27T20:54:41.338631480Z   File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
2025-01-27T20:54:41.338650079Z     return await old_call(app, scope, new_receive, new_send, **kwargs)
2025-01-27T20:54:41.338657030Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.338688657Z   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/authentication.py", line 48, in __call__
2025-01-27T20:54:41.338694435Z     await self.app(scope, receive, send)
2025-01-27T20:54:41.338700053Z   File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
2025-01-27T20:54:41.338765017Z     return await old_call(app, scope, new_receive, new_send, **kwargs)
2025-01-27T20:54:41.338777917Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.338782992Z   File "/backend/handler/auth/middleware.py", line 28, in __call__
2025-01-27T20:54:41.338789255Z     await super().__call__(scope, receive, send)
2025-01-27T20:54:41.338808369Z   File "/src/.venv/lib/python3.12/site-packages/starlette_csrf/middleware.py", line 72, in __call__
2025-01-27T20:54:41.338814204Z     await self.app(scope, receive, send)
2025-01-27T20:54:41.338820048Z   File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
2025-01-27T20:54:41.338842518Z     return await old_call(app, scope, new_receive, new_send, **kwargs)
2025-01-27T20:54:41.338847602Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.338852837Z   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/cors.py", line 85, in __call__
2025-01-27T20:54:41.338857939Z     await self.app(scope, receive, send)
2025-01-27T20:54:41.338876587Z   File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 298, in _sentry_exceptionmiddleware_call
2025-01-27T20:54:41.338881729Z     await old_call(self, scope, receive, send)
2025-01-27T20:54:41.338886496Z   File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
2025-01-27T20:54:41.338894873Z     return await old_call(app, scope, new_receive, new_send, **kwargs)
2025-01-27T20:54:41.338899755Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.338904854Z   File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
2025-01-27T20:54:41.338915869Z     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
2025-01-27T20:54:41.338921408Z   File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
2025-01-27T20:54:41.338926263Z     raise exc
2025-01-27T20:54:41.338930757Z   File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
2025-01-27T20:54:41.338938219Z     await app(scope, receive, sender)
2025-01-27T20:54:41.338941289Z   File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 715, in __call__
2025-01-27T20:54:41.338944604Z     await self.middleware_stack(scope, receive, send)
2025-01-27T20:54:41.338948253Z   File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 735, in app
2025-01-27T20:54:41.338956206Z     await route.handle(scope, receive, send)
2025-01-27T20:54:41.338961810Z   File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 288, in handle
2025-01-27T20:54:41.338967879Z     await self.app(scope, receive, send)
2025-01-27T20:54:41.338983380Z   File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 76, in app
2025-01-27T20:54:41.338989019Z     await wrap_app_handling_exceptions(app, request)(scope, receive, send)
2025-01-27T20:54:41.338993995Z   File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
2025-01-27T20:54:41.338998845Z     raise exc
2025-01-27T20:54:41.339104354Z   File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
2025-01-27T20:54:41.339126697Z     await app(scope, receive, sender)
2025-01-27T20:54:41.339137765Z   File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 73, in app
2025-01-27T20:54:41.339191781Z     response = await f(request)
2025-01-27T20:54:41.339200860Z                ^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.339205190Z   File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/fastapi.py", line 143, in _sentry_app
2025-01-27T20:54:41.339209844Z     return await old_app(*args, **kwargs)
2025-01-27T20:54:41.339215157Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.339229666Z   File "/src/.venv/lib/python3.12/site-packages/fastapi/routing.py", line 301, in app
2025-01-27T20:54:41.339234264Z     raw_response = await run_endpoint_function(
2025-01-27T20:54:41.339238407Z                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.339243376Z   File "/src/.venv/lib/python3.12/site-packages/fastapi/routing.py", line 212, in run_endpoint_function
2025-01-27T20:54:41.339257557Z     return await dependant.call(**values)
2025-01-27T20:54:41.339262395Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.339267315Z   File "/backend/endpoints/auth.py", line 228, in login_via_openid
2025-01-27T20:54:41.339277237Z     return await oauth.openid.authorize_redirect(request, OIDC_REDIRECT_URI)
2025-01-27T20:54:41.339283199Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.339341815Z   File "/src/.venv/lib/python3.12/site-packages/authlib/integrations/starlette_client/apps.py", line 34, in authorize_redirect
2025-01-27T20:54:41.339350921Z     rv = await self.create_authorization_url(redirect_uri, **kwargs)
2025-01-27T20:54:41.339380834Z          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.339388179Z   File "/src/.venv/lib/python3.12/site-packages/authlib/integrations/base_client/async_app.py", line 95, in create_authorization_url
2025-01-27T20:54:41.339392756Z     metadata = await self.load_server_metadata()
2025-01-27T20:54:41.339458498Z                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.339470988Z   File "/src/.venv/lib/python3.12/site-packages/authlib/integrations/base_client/async_app.py", line 76, in load_server_metadata
2025-01-27T20:54:41.339476914Z     resp = await client.request('GET', self._server_metadata_url, withhold_token=True)
2025-01-27T20:54:41.339496112Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.339501014Z   File "/src/.venv/lib/python3.12/site-packages/authlib/integrations/httpx_client/oauth2_client.py", line 90, in request
2025-01-27T20:54:41.339506137Z     return await super().request(
2025-01-27T20:54:41.339510915Z            ^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.339527587Z   File "/src/.venv/lib/python3.12/site-packages/httpx/_client.py", line 1585, in request
2025-01-27T20:54:41.339532904Z     return await self.send(request, auth=auth, follow_redirects=follow_redirects)
2025-01-27T20:54:41.339537544Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.339550699Z   File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/httpx.py", line 142, in send
2025-01-27T20:54:41.339555508Z     rv = await real_send(self, request, **kwargs)
2025-01-27T20:54:41.339560101Z          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.339564917Z   File "/src/.venv/lib/python3.12/site-packages/httpx/_client.py", line 1674, in send
2025-01-27T20:54:41.339575722Z     response = await self._send_handling_auth(
2025-01-27T20:54:41.339580588Z                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.339585235Z   File "/src/.venv/lib/python3.12/site-packages/httpx/_client.py", line 1702, in _send_handling_auth
2025-01-27T20:54:41.339599559Z     response = await self._send_handling_redirects(
2025-01-27T20:54:41.339604663Z                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.339609565Z   File "/src/.venv/lib/python3.12/site-packages/httpx/_client.py", line 1739, in _send_handling_redirects
2025-01-27T20:54:41.339615084Z     response = await self._send_single_request(request)
2025-01-27T20:54:41.339625396Z                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.339632075Z   File "/src/.venv/lib/python3.12/site-packages/httpx/_client.py", line 1776, in _send_single_request
2025-01-27T20:54:41.339637402Z     response = await transport.handle_async_request(request)
2025-01-27T20:54:41.339642568Z                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.339656651Z   File "/src/.venv/lib/python3.12/site-packages/httpx/_transports/default.py", line 376, in handle_async_request
2025-01-27T20:54:41.339671890Z     with map_httpcore_exceptions():
2025-01-27T20:54:41.339680479Z          ^^^^^^^^^^^^^^^^^^^^^^^^^
2025-01-27T20:54:41.339685403Z   File "/usr/lib/python3.12/contextlib.py", line 158, in __exit__
2025-01-27T20:54:41.339699422Z     self.gen.throw(value)
2025-01-27T20:54:41.339704275Z   File "/src/.venv/lib/python3.12/site-packages/httpx/_transports/default.py", line 89, in map_httpcore_exceptions
2025-01-27T20:54:41.339709414Z     raise mapped_exc(message) from exc
2025-01-27T20:54:41.339715503Z httpx.UnsupportedProtocol: Request URL is missing an 'http://' or 'https://' protocol
```

```
services:
  romm:
    image: rommapp/romm:latest
    container_name: romm
    restart: unless-stopped
    profiles: ["all", "romm"]
    networks:
      - ${PROXY_NETWORK}
      - ${APP_NETWORK}
    volumes:
      - ${APPDATA_DIR}/romm/assets:/romm/assets
      - ${APPDATA_DIR}/romm/config:/romm/config
      - ${APPDATA_DIR}/romm/redis:/redis-data
      - ${APPDATA_DIR}/romm/resources:/romm/resources
      - ${GAMES_DIR}/library:/romm/library
    environment:
      TZ: ${TZ}
      DB_HOST: romm-db
      DB_NAME: ${ROMM_MYSQL_DB}
      DB_USER: ${ROMM_MYSQL_USER}
      DB_PASSWD: ${ROMM_MYSQL_PASSWORD}
      ROMM_AUTH_SECRET_KEY: ${ROMM_AUTH_KEY}
      IGDB_CLIENT_ID: ${ROMM_IGDB_ID}
      IGDB_CLIENT_SECRET: ${ROMM_IGDB_SECRET}
      MOBYGAMES_API_KEY: ${ROMM_MOBY_KEY}
      STEAMGRIDDB_API_KEY: ${ROMM_SGDB_KEY}
      LOG_LEVEL: debug
      OIDC_ENABLED: true
      OIDC_PROVIDER: authentik
      OIDC_CLIENT_ID: ${ROMM_OIDC_ID}
      OIDC_CLIENT_SECRET: ${ROMM_OIDC_SECRET}
      OIDC_REDIRECT:URI: https://${SUB_ROMM}.${DOMAINNAME}/api/oauth/openid
      OIDC_SERVER_APPLICATION: https://${SUB_LOGIN}.${DOMAINNAME}/application/o/romm
    labels:
      ## Logging
      logging.promtail: true
      ## Traefik
      traefik.enable: true
      traefik.docker.network: ${PROXY_NETWORK} # For containers on multiple bridge networks, force the correct proxy network
      traefik.external.cname: true # For services that need a public cname, not only private access
      traefik.http.routers.romm.entrypoints: https
      traefik.http.routers.romm.rule: Host(`${SUB_ROMM}.${DOMAINNAME}`)
      traefik.http.routers.romm.middlewares: chain-base@file # Change to chain-no-auth if using OIDC in most cases
      traefik.http.routers.romm.service: romm # Not required if only one service is being defined - https://doc.traefik.io/traefik/routing/providers/docker/#service-definition
      traefik.http.services.romm.loadbalancer.server.port: 8080 # May not be required if only one port is exposed by default - https://doc.traefik.io/traefik/providers/docker/#port-detection
      ## Unraid
      net.unraid.docker.managed: composeman
      net.unraid.docker.webui: https://${SUB_ROMM}.${DOMAINNAME}
      net.unraid.docker.icon: 
      ## Homepage
      homepage.instance.me.group: Gaming
      homepage.instance.me.name: ROM Manager
      homepage.instance.me.icon: # Can use prebuild icon library, or “mdi-” "si-" "sh-" prebuilt (https://pictogrammers.com/library/mdi/, https://simpleicons.org, https://selfh.st/icons)
      homepage.instance.me.href: https://${SUB_ROMM}.${DOMAINNAME}
      homepage.instance.me.weight: 
      homepage.instance.me.description: Roms
      ## Autokuma
      kuma.__app: '{ "name": "Romm", "type": "web-group", "url": "https://${SUB_ROMM}.${DOMAINNAME}", "internal_port": "8080" }'

  romm-db:
    image: lscr.io/linuxserver/mariadb:latest
    container_name: romm-db
    restart: always
    profiles: ["all", "romm"]
    networks:
      - ${APP_NETWORK}
    volumes:
      - ${APPDATA_DIR}/romm/db:/config
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
    environment:
      PUID: ${PUID}
      PGID: ${PGID}
      MYSQL_ROOT_PASSWORD: ${ROMM_MYSQL_ROOT_PASSWORD}
      MYSQL_DATABASE: ${ROMM_MYSQL_DB}
      MYSQL_USER: ${ROMM_MYSQL_USER}
      MYSQL_PASSWORD: ${ROMM_MYSQL_PASSWORD}
    labels:
      ## Logging
      logging.promtail: true
      ## Unraid
      net.unraid.docker.managed: composeman
      net.unraid.docker.icon: 
      ## Autokuma
      kuma.__app: '{ "name": "Romm MySQL", "type": "mysql", "service": "Romm", "db_url": "mysql://${ROMM_MYSQL_USER}:${ROMM_MYSQL_PASSWORD}@romm-db:3306" }'
```

I tried pivoting to Pocket ID and was following the guide at [the wiki](https://github.com/rommapp/romm/wiki/OIDC-Setup-With-PocketID), but I received the following error about the redirect URL having a bad format:

Config:
```
      OIDC_ENABLED: true
      OIDC_PROVIDER: pocketid
      OIDC_CLIENT_ID: ${ROMM_OIDC_ID}
      OIDC_CLIENT_SECRET: ${ROMM_OIDC_SECRET}
      OIDC_REDIRECT_URI: https://${SUB_ROMM}.${DOMAINNAME}/api/oauth/openid
      OIDC_SERVER_APPLICATION: https://${SUB_POCKET_ID}.${DOMAINNAME}/authorize
```

```
[2025-02-19 21:54:47 -0800] [25] [ERROR] Exception in ASGI application
Traceback (most recent call last):
  File "/src/.venv/lib/python3.12/site-packages/uvicorn/protocols/http/h11_impl.py", line 407, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/uvicorn/middleware/proxy_headers.py", line 69, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 409, in _sentry_patched_asgi_app
    return await middleware(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/asgi.py", line 161, in _run_asgi3
    return await self._run_app(scope, receive, send, asgi_version=3)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/asgi.py", line 262, in _run_app
    raise exc from None
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/asgi.py", line 257, in _run_app
    return await self.app(
           ^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/applications.py", line 113, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/errors.py", line 187, in __call__
    raise exc
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/errors.py", line 165, in __call__
    await self.app(scope, receive, _send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 185, in __call__
    with collapse_excgroups():
         ^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/contextlib.py", line 158, in __exit__
    self.gen.throw(value)
  File "/src/.venv/lib/python3.12/site-packages/starlette/_utils.py", line 82, in collapse_excgroups
    raise exc
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 187, in __call__
    response = await self.dispatch_func(request, call_next)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/backend/utils/context.py", line 41, in set_context_middleware
    return await call_next(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 163, in call_next
    raise app_exc
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 149, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/backend/handler/auth/middleware.py", line 156, in __call__
    await self.app(scope, receive, send_wrapper)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 348, in _sentry_authenticationmiddleware_call
    await old_call(self, scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/authentication.py", line 48, in __call__
    await self.app(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/backend/handler/auth/middleware.py", line 28, in __call__
    await super().__call__(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/starlette_csrf/middleware.py", line 72, in __call__
    await self.app(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/cors.py", line 85, in __call__
    await self.app(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 298, in _sentry_exceptionmiddleware_call
    await old_call(self, scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call
    return await old_call(app, scope, new_receive, new_send, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 715, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 735, in app
    await route.handle(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 73, in app
    response = await f(request)
               ^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/fastapi.py", line 143, in _sentry_app
    return await old_app(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/fastapi/routing.py", line 301, in app
    raw_response = await run_endpoint_function(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/fastapi/routing.py", line 212, in run_endpoint_function
    return await dependant.call(**values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/backend/endpoints/auth.py", line 228, in login_via_openid
    return await oauth.openid.authorize_redirect(request, OIDC_REDIRECT_URI)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/authlib/integrations/starlette_client/apps.py", line 34, in authorize_redirect
    rv = await self.create_authorization_url(redirect_uri, **kwargs)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/authlib/integrations/base_client/async_app.py", line 95, in create_authorization_url
    metadata = await self.load_server_metadata()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/authlib/integrations/base_client/async_app.py", line 76, in load_server_metadata
    resp = await client.request('GET', self._server_metadata_url, withhold_token=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/authlib/integrations/httpx_client/oauth2_client.py", line 90, in request
    return await super().request(
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/httpx/_client.py", line 1585, in request
    return await self.send(request, auth=auth, follow_redirects=follow_redirects)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/httpx.py", line 142, in send
    rv = await real_send(self, request, **kwargs)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/httpx/_client.py", line 1674, in send
    response = await self._send_handling_auth(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/httpx/_client.py", line 1702, in _send_handling_auth
    response = await self._send_handling_redirects(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/httpx/_client.py", line 1739, in _send_handling_redirects
    response = await self._send_single_request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/httpx/_client.py", line 1776, in _send_single_request
    response = await transport.handle_async_request(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/src/.venv/lib/python3.12/site-packages/httpx/_transports/default.py", line 376, in handle_async_request
    with map_httpcore_exceptions():
         ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/contextlib.py", line 158, in __exit__
    self.gen.throw(value)
  File "/src/.venv/lib/python3.12/site-packages/httpx/_transports/default.py", line 89, in map_httpcore_exceptions
    raise mapped_exc(message) from exc
httpx.UnsupportedProtocol: Request URL is missing an 'http://' or 'https://' protocol.
```

Also wanted to chime in to say I'm experiencing similar issues with Entra ID/AAD. Authorization pops up, I sign in with ms account but then I get internal server error. I see in url that the code is passed along (https://rommurl/api/oauth/openid?code=...)

My setup:
```
      - OIDC_ENABLED=true
      - OIDC_PROVIDER=microsoft
      - OIDC_CLIENT_ID=
      - OIDC_CLIENT_SECRET=
      - OIDC_REDIRECT_URI=https://<rommurl>/api/oauth/openid
      - OIDC_SERVER_APPLICATION_URL=https://login.microsoftonline.com/<tenant id>/v2.0
      - LOGLEVEL=debug
```


Debug log:
```
[2025-03-04 20:04:29 +0000] [24] [ERROR] Exception in ASGI application

Traceback (most recent call last):

  File "/src/.venv/lib/python3.12/site-packages/uvicorn/protocols/http/h11_impl.py", line 407, in run_asgi

    result = await app(  # type: ignore[func-returns-value]

             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/src/.venv/lib/python3.12/site-packages/uvicorn/middleware/proxy_headers.py", line 69, in __call__

    return await self.app(scope, receive, send)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/src/.venv/lib/python3.12/site-packages/fastapi/applications.py", line 1054, in __call__

    await super().__call__(scope, receive, send)

  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 409, in _sentry_patched_asgi_app

    return await middleware(scope, receive, send)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/asgi.py", line 161, in _run_asgi3

    return await self._run_app(scope, receive, send, asgi_version=3)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/asgi.py", line 262, in _run_app

    raise exc from None

  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/asgi.py", line 257, in _run_app

    return await self.app(

           ^^^^^^^^^^^^^^^

  File "/src/.venv/lib/python3.12/site-packages/starlette/applications.py", line 113, in __call__

    await self.middleware_stack(scope, receive, send)

  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call

    return await old_call(app, scope, new_receive, new_send, **kwargs)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/errors.py", line 187, in __call__

    raise exc

  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/errors.py", line 165, in __call__

    await self.app(scope, receive, _send)

  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call

    return await old_call(app, scope, new_receive, new_send, **kwargs)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 185, in __call__

    with collapse_excgroups():

         ^^^^^^^^^^^^^^^^^^^^

  File "/usr/lib/python3.12/contextlib.py", line 158, in __exit__

    self.gen.throw(value)

  File "/src/.venv/lib/python3.12/site-packages/starlette/_utils.py", line 82, in collapse_excgroups

    raise exc

  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 187, in __call__

    response = await self.dispatch_func(request, call_next)

               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/backend/utils/context.py", line 41, in set_context_middleware

    return await call_next(request)

           ^^^^^^^^^^^^^^^^^^^^^^^^

  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 163, in call_next

    raise app_exc

  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/base.py", line 149, in coro

    await self.app(scope, receive_or_disconnect, send_no_error)

  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call

    return await old_call(app, scope, new_receive, new_send, **kwargs)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/backend/handler/auth/middleware.py", line 156, in __call__

    await self.app(scope, receive, send_wrapper)

  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 348, in _sentry_authenticationmiddleware_call

    await old_call(self, scope, receive, send)

  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call

    return await old_call(app, scope, new_receive, new_send, **kwargs)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/authentication.py", line 48, in __call__

    await self.app(scope, receive, send)

  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call

    return await old_call(app, scope, new_receive, new_send, **kwargs)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/backend/handler/auth/middleware.py", line 28, in __call__

    await super().__call__(scope, receive, send)

  File "/src/.venv/lib/python3.12/site-packages/starlette_csrf/middleware.py", line 72, in __call__

    await self.app(scope, receive, send)

  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call

    return await old_call(app, scope, new_receive, new_send, **kwargs)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/cors.py", line 85, in __call__

    await self.app(scope, receive, send)

  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 298, in _sentry_exceptionmiddleware_call

    await old_call(self, scope, receive, send)

  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/starlette.py", line 200, in _create_span_call

    return await old_call(app, scope, new_receive, new_send, **kwargs)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/src/.venv/lib/python3.12/site-packages/starlette/middleware/exceptions.py", line 62, in __call__

    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)

  File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app

    raise exc

  File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app

    await app(scope, receive, sender)

  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 715, in __call__

    await self.middleware_stack(scope, receive, send)

  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 735, in app

    await route.handle(scope, receive, send)

  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 288, in handle

    await self.app(scope, receive, send)

  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 76, in app

    await wrap_app_handling_exceptions(app, request)(scope, receive, send)

  File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app

    raise exc

  File "/src/.venv/lib/python3.12/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app

    await app(scope, receive, sender)

  File "/src/.venv/lib/python3.12/site-packages/starlette/routing.py", line 73, in app

    response = await f(request)

               ^^^^^^^^^^^^^^^^

  File "/src/.venv/lib/python3.12/site-packages/sentry_sdk/integrations/fastapi.py", line 143, in _sentry_app

    return await old_app(*args, **kwargs)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/src/.venv/lib/python3.12/site-packages/fastapi/routing.py", line 301, in app

    raw_response = await run_endpoint_function(

                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/src/.venv/lib/python3.12/site-packages/fastapi/routing.py", line 212, in run_endpoint_function

    return await dependant.call(**values)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/backend/endpoints/auth.py", line 254, in auth_openid

    token = await oauth.openid.authorize_access_token(request)

            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/src/.venv/lib/python3.12/site-packages/authlib/integrations/starlette_client/apps.py", line 80, in authorize_access_token

    params = self._format_state_params(state_data, params)

             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/src/.venv/lib/python3.12/site-packages/authlib/integrations/base_client/sync_app.py", line 234, in _format_state_params

    raise MismatchingStateError()

authlib.integrations.base_client.errors.MismatchingStateError: mismatching_state: CSRF Warning! State not equal in request and response.
```

--- a/backend/handler/auth/base_handler.py
+++ b/backend/handler/auth/base_handler.py
@@ -317,14 +317,14 @@ class OpenIDHandler:
                 "User with email '%s' not found, creating new user",
                 hl(email, color=CYAN),
             )
-            user = User(
+            new_user = User(
                 username=preferred_username,
                 hashed_password=str(uuid.uuid4()),
                 email=email,
                 enabled=True,
                 role=Role.VIEWER,
             )
-            db_user_handler.add_user(user)
+            user = db_user_handler.add_user(new_user)
 
         if not user.enabled:
             raise UserDisabledException

